internal/proxyservice/proxy_service.go

```diff
@@ -65,6 +66,8 @@ func (s *proxyServices) servicesToStart() (servicesToStart []internal.Service) {
 	//   hardcoded dep on an impl type of v2.  All deps need to be injected.
 	configsByType := v2.NewConfigsByType(s.config.Services, s.availPlugins)
 	httpPlugins := s.availPlugins.HTTPPlugins()
+	// TODO: maybe this abstraction doesn't work ?
```

Comment:
Fully agreed - this "configsByType" seems very geared toward copy/pasting blocks of code

Reply:
this is the stuff to be deferred to a later date?

Reply:
yup. please open an issue for it though.